fapolicy_analyzer/ui/faprofiler.py

```diff
@@ -37,17 +40,68 @@ class ProfSessionArgsStatus(Enum):
     OK = 0
     EXEC_EMPTY = 1
     EXEC_DOESNT_EXIST = 2
-    EXEC_NOT_EXECUTABLE = 3
-    USER_DOESNT_EXIST = 4
-    PWD_DOESNT_EXIST = 5
-    PWD_ISNT_DIR = 6
-    UNKNOWN = 7
+    EXEC_NOT_EXEC = 3
+    EXEC_NOT_FOUND = 4
+    USER_DOESNT_EXIST = 5
+    PWD_DOESNT_EXIST = 6
+    PWD_ISNT_DIR = 7,
+    ENV_VARS_FORMATING = 8,
+    UNKNOWN = 9
 
 
+# ########################## Utility Functions ###########################
 def EnumErrorPairs2Str(dictStatusEnums):
+    """Converts the dict collection of profiler argument error to a string for
+    displaying in the pop-up notification.
+    """
     return "\n  ".join([f"Error: {r}" for r in (dictStatusEnums or {}).values()])
 
 
+def expand_path(colon_separated_str, cwd="."):
+    """Expands user supplied PATH argument that contains embedded 'PATH'
+    variable, leading colon, terminating colon, or an intermediate double colon.
+    These final three colon notations map to the user supplied working dir.
+    """
+    logging.debug(f"expand_path({colon_separated_str}, {cwd})")
+
+    # Expand native PATH env var if in user provided PATH env var string
+    regexPath = re.compile(r"\$\{?PATH\}?")
+
+    expanded_path = regexPath.sub(os.environ.get("PATH"), colon_separated_str)
+
+    # Expand implied '.' notation to supplied cwd argument in search PATH
+    regexTrailingColon = re.compile(r":$")
+    regexLeadingColon = re.compile(r"^:")
+    regexDoubleColon = re.compile(r"::")
+
+    print(f"-> {expanded_path}")  # ToDo: Remove prior to push
+    if regexTrailingColon.search(expanded_path):
+        logging.debug("Trailing colon detected.")
+        expanded_path = regexTrailingColon.sub(":.", expanded_path)
+    if regexLeadingColon.search(expanded_path):
+        logging.debug("Leading colon detected.")
+        expanded_path = regexLeadingColon.sub(".:", expanded_path)
+    if regexDoubleColon.search(expanded_path):
+        logging.debug("Internal double colon detected")
+        expanded_path = regexDoubleColon.sub(":.:", expanded_path)
+
+    # Expand periods with user supplied absolute path if provided otherwise
+    # use cwd from runtime environment
+    if cwd != ".":
+        expanded_path = re.sub(r":\.$", f":{cwd}", expanded_path)
+        expanded_path = re.sub(r"^\.:", f"{cwd}:", expanded_path)
+        expanded_path = re.sub(r":\.:", f":{cwd}:", expanded_path)
+        expanded_path = re.sub(r"^\.$", f"{cwd}", expanded_path)
```

Comment:
I haven't tested this but I think you could replace this entire chunk with these lines below and end up with the same thing.

```suggestion
    expanded_path = re.sub(r":\.{0,1}$", f":{cwd}", expanded_path)
    expanded_path = re.sub(r"^\.{0,1}:", f"{cwd}:, expanded_path)
    expanded_path = re.sub(r":\.{0,1}:", "f":{cwd}:"", expanded_path)
    expanded_path = re.sub(r"^\.$", f"{cwd}", expanded_path)
```

Reply:
Seems to work fine. I'll test it more on an rpm install platform. Thank you.